spyder/plugins/editor/plugin.py

```diff
@@ -287,27 +287,29 @@ def restore_scrollbar_position(self):
     def report_open_file(self, options):
         """Request to start a LSP server to attend a language."""
         filename = options['filename']
-        logger.debug('Call LSP for %s' % filename)
         language = options['language']
+        logger.debug('Call LSP for %s [%s]' % (filename, language))
         callback = options['codeeditor']
         stat = self.main.lspmanager.start_client(language.lower())
         self.main.lspmanager.register_file(
             language.lower(), filename, callback)
         if stat:
             if language.lower() in self.lsp_editor_settings:
+                logger.debug('{0} LSP is ready'.format(language))
                 self.lsp_server_ready(
                     language.lower(), self.lsp_editor_settings[
                         language.lower()])
             else:
-                editor = self.get_current_editor()
-                editor.lsp_ready = False
+                if callback.language == language.lower():
+                    logger.debug('Setting {0} LSP off'.filename)

```

Comment:
This is wrong. It needs to be

    logger.debug('Setting {0} LSP off'.format(filename))